### ISSUE TITLE
We need a common SQL schema for the scrapers.

### ISSUE THREAD

**Issue by kremio** [tatortrechts/data.tatortrechts.de#20]
We need a common database schema to make it easier to write tools that work with all scrapers.

I'm in favour of using separated tables for the sources and the locations, using the report URI as a foreign key:
CREATE TABLE "data" ("uri" VARCHAR  PRIMARY KEY, "title" TEXT  , "description" TEXT  NOT NULL, "startDate" datetime  NOT NULL, "endDate" datetime  , "iso3166_2" VARCHAR  NOT NULL, "motives" TEXT  , "contexts" TEXT  , "factums" TEXT  , "tags" TEXT  , "createdDate" datetime  NOT NULL);

CREATE TABLE "source" ("id" INTEGER  PRIMARY KEY AUTOINCREMENT, "reportURI" VARCHAR  NOT NULL, "name" VARCHAR  NOT NULL, "publishedDate" datetime  , "url" VARCHAR  , "createdDate" datetime  NOT NULL);

CREATE TABLE "location" ("id" INTEGER  PRIMARY KEY AUTOINCREMENT, "reportURI" VARCHAR  NOT NULL, "subdivisions" TEXT  NOT NULL, "latitude" REAL  , "longitude" REAL  , "createdDate" datetime  NOT NULL);

Another option would be to have a single table, with locations and sources being single columns storing stringified JSON.

I personally prefer having separate tables, because:
- it is the most natural mapping between the JSON definition and SQL.
- it makes it easy to check that the scraper is parsing the pages correctly.
- it provides some amount of type checking

**Comment by kremio** [comment]
I did some testing with morph.io and it is possible to run python (and probably ruby) code from a Node.js project.
As I have already written a library that sets up the sqlite database and validate documents before inserting them, it could be interesting to re-use this code in other scrapers.
I implemented this solution here: https://github.com/kremio/morph-io-test
The idea is that the python scraping code (https://github.com/kremio/morph-io-test/blob/master/json-gen.py in the example) is executed from the node.js runtime and its output (on stdout) is piped to a node.js stream who handles the db inserts. The python script is supposed to output scraped reports as JSON following the format defined in https://github.com/codeforberlin/right-wing-violence-de/blob/master/docs/ReportFormat.md
Note that the python code MUST flush its stdout between each report, otherwise something horrible will happen.
